From: Chip Coldwell <coldwell@redhat.com>
Subject: [PATCH RHEL5.1]: BZ#220603 Missing PCI Device in aic79xx driver
Date: Wed, 17 Jan 2007 13:38:25 -0500 (EST)
Bugzilla: 220603
Message-Id: <Pine.LNX.4.64.0701171335060.18888@localhost.localdomain>
Changelog: [scsi] Missing PCI Device in aic79xx driver

A small patch to add support for another device ID to the Adaptec
aic79xx SCSI driver.

0. Patch: inline and attached
1. Risk: low.
2. KABI: none.
3. Upstream: yes

commit d32adcb85c74fd81963714689842993e7014515f
Author: Salyzyn, Mark <mark_salyzyn@adaptec.com>

    [SCSI] aic79xx: Add ASC-29320LPE ids to driver

    Simple patch to add the new PCIe version of the 29320 card.

    Signed-off: Mark Salyzyn <Mark_Salyzyn@adaptec.com>
    Signed-off-by: James Bottomley <James.Bottomley@SteelEye.com>

4. Testing: it compiles.  Don't have the hardware to verify that it
   works.

--- a/drivers/scsi/aic7xxx/aic79xx_osm_pci.c	2006-12-05 
07:25:59.652815006 -0500
+++ b/drivers/scsi/aic7xxx/aic79xx_osm_pci.c	2006-12-05 
07:28:25.288658410 -0500
@@ -62,6 +62,7 @@
 	/* aic7901 based controllers */
 	ID(ID_AHA_29320A),
 	ID(ID_AHA_29320ALP),
+	ID(ID_AHA_29320LPE),
 	/* aic7902 based controllers */
 	ID(ID_AHA_29320),
 	ID(ID_AHA_29320B),
--- a/drivers/scsi/aic7xxx/aic79xx_pci.c	2006-12-05 
07:26:15.541834499 -0500
+++ b/drivers/scsi/aic7xxx/aic79xx_pci.c	2006-12-05 
07:35:26.469105302 -0500
@@ -109,7 +109,13 @@
 	{
 		ID_AHA_29320ALP,
 		ID_ALL_MASK,
-		"Adaptec 29320ALP Ultra320 SCSI adapter",
+		"Adaptec 29320ALP PCIx Ultra320 SCSI adapter",
+		ahd_aic7901_setup
+	},
+	{
+		ID_AHA_29320LPE,
+		ID_ALL_MASK,
+		"Adaptec 29320LPE PCIe Ultra320 SCSI adapter",
 		ahd_aic7901_setup
 	},
 	/* aic7901A based controllers */
--- a/drivers/scsi/aic7xxx/aic79xx_pci.h	2006-12-05 
07:24:36.263207576 -0500
+++ b/drivers/scsi/aic7xxx/aic79xx_pci.h	2006-12-05 
07:25:22.825405020 -0500
@@ -51,6 +51,7 @@
 #define ID_AIC7901			0x800F9005FFFF9005ull
 #define ID_AHA_29320A			0x8000900500609005ull
 #define ID_AHA_29320ALP			0x8017900500449005ull
+#define ID_AHA_29320LPE			0x8017900500459005ull

 #define ID_AIC7901A			0x801E9005FFFF9005ull
 #define ID_AHA_29320LP			0x8014900500449005ull


